simple/directed.go

```diff
@@ -120,7 +120,6 @@ func (g *DirectedGraph) RemoveNode(n graph.Node) {
 	}
 	delete(g.predecessors, n.ID())
 
-	g.maxID-- // Fun facts: even if this ID doesn't exist this still works!
```

Comment:
It seems to me that this should be done, but conditionally on the removed node having n.ID() == g.maxID. Please do that - dropping the comment.